controllers/front/OrderController.php

```diff
@@ -145,7 +148,12 @@ private function restorePersistedData(CheckoutProcess $process)
         if (!is_array($data)) {
             $data = [];
         }
-        $process->restorePersistedData($data);
+
+        $cartChecksum = new CartChecksum();
+        $checksum = $cartChecksum->checksum($this->context->cart);
```

Comment:
I like the idea :) if the checksum is only specific to CartCore, we could probably add this function to this class.

```php
$cart = new CartCore();
$cart->generateChecksum();
// you have populated a new property `checksum` with a checksum

// OR

$cart = new CartCore(); // and on instanciation we generate checksum ..
```

What do you think ?